src/zenml/repository.py

```diff
@@ -113,7 +114,16 @@ def __init__(self, root: Optional[Path] = None):
 
         Raises:
             RepositoryNotFoundError: If no ZenML repository directory is found.
+            RuntimeError: If trying to create a `Repository` instance while a
+                ZenML step is being executed.
         """
+        if Environment.currently_running_step():
+            raise RuntimeError(
+                "Unable to access repository during step execution. If you "
+                "require access to the artifact or metadata store, please use "
+                "a `StepContext` inside your step instead."
```

Comment:
Personally, I think we could add a tiny example here that tells the user how he can achieve the desired fucntionality. Alternatively we could also refer them to the corresponding docs: https://docs.zenml.io/guides/index/historic-runs#utilizing-stepcontext

Reply:
That is a great idea, I think I'd probably go for the URL as that goes into way more detail. Now I just need an Exception name :D

Reply:
How about ForbiddenRepositoryAccessError ?

Reply:
Fixed